sdk/search/Azure.AI.Search/src/Generated/Operations/DocumentsOperations.cs

```diff
@@ -0,0 +1,596 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.AI.Search.Models;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.AI.Search
+{
+    internal partial class DocumentsOperations
+    {
+        private string endpoint;
+        private string indexName;
+        private string ApiVersion;
+        private ClientDiagnostics clientDiagnostics;
+        private HttpPipeline pipeline;
+        /// <summary> Initializes a new instance of DocumentsOperations. </summary>
+        public DocumentsOperations(ClientDiagnostics clientDiagnostics, HttpPipeline pipeline, string endpoint, string indexName, string ApiVersion = "2019-05-06")
+        {
+            if (endpoint == null)
+            {
+                throw new ArgumentNullException(nameof(endpoint));
+            }
+            if (indexName == null)
+            {
+                throw new ArgumentNullException(nameof(indexName));
+            }
+            if (ApiVersion == null)
+            {
+                throw new ArgumentNullException(nameof(ApiVersion));
+            }
+
+            this.endpoint = endpoint;
+            this.indexName = indexName;
+            this.ApiVersion = ApiVersion;
+            this.clientDiagnostics = clientDiagnostics;
+            this.pipeline = pipeline;
+        }
+        internal HttpMessage CreateCountRequest(Guid? clientRequestId)
+        {
+            var message = pipeline.CreateMessage();
+            var request = message.Request;
+            request.Method = RequestMethodAdditional.Get;
+            var uri = new RawRequestUriBuilder();
+            uri.AppendRaw(endpoint, false);
+            uri.AppendRaw("/indexes('", false);
+            uri.AppendRaw(indexName, true);
+            uri.AppendRaw("')", false);
+            uri.AppendPath("/docs/$count", false);
+            uri.AppendQuery("api-version", ApiVersion, true);
+            request.Uri = uri;
+            if (clientRequestId != null)
+            {
+                request.Headers.Add("client-request-id", clientRequestId.Value);
+            }
+            return message;
+        }
+        /// <summary> Queries the number of documents in the index. </summary>
+        /// <param name="clientRequestId"> The tracking ID sent with the request to help with debugging. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        public async ValueTask<Response<long>> CountAsync(Guid? clientRequestId, CancellationToken cancellationToken = default)
+        {
+
+            using var scope = clientDiagnostics.CreateScope("DocumentsOperations.Count");
```

Comment:
@pakrym - At some point in the not too distant future it'd be nice if we could customize these so I don't have to put [ForwardsClientCalls] on every public operation method.